clues.go

```diff
@@ -26,19 +28,29 @@ import (
 // declared by an ancestor, the child's entry takes priority.
 type dataNode struct {
 	parent *dataNode
+	id     string
 	vs     map[string]any
 }
 
+func makeNodeID() string {
+	return uuid.NewString()[:8]
```

Comment:
I'm unsure if this will be sufficiently unique for our purposes across machines. The google uuid package page says that it's based on RFC 4122, which [defines](https://datatracker.ietf.org/doc/html/rfc4122#section-4.1.2) the byte ordering for UUIDs. From the RFC, it seems that octets 0-3 are timestamp based, but it's unclear what ordering they use. If they use big endian, it's possible that pulling the 0th octet won't yield sufficient change if we're making node IDs with little time between them because it will use high(er) order bits in the timestamp

Reply:
hmm, fair. I might swap that out for the final 8 chars.  Or maybe a mix of both.

Overall I'm not entirely worried about this, because the more important part of the behavior is chaining ids to represent a process branch.  If two processes produce colliding ids, that's only a problem if they share the same parent node.

Reply:
maybe you could extract the UUIDs [time](https://pkg.go.dev/github.com/google/uuid#UUID.Time) and pull out the lower digits?